### ISSUE TITLE
ZeroSegments error

### ISSUE THREAD

**Issue by leszko** [livepeer/go-livepeer#2322]
The `ZeroSegments` error was spotted in the Orchestrator's logs ([Discord discussion](https://discord.com/channels/423160867534929930/429350403272998913/953677818758311946)).

```
I0316 08:06:03.320533    2136 segment_rpc.go:100] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Received segment dur=3.041s
I0316 08:06:03.320533    2136 census.go:1136] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentEmerged... duration=3.041
I0316 08:06:03.320533    2136 orchestrator.go:170] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Receiving ticket sessionID=3c256dbc faceValue=0.05242065252 ETH winProb=0.0000190765 ev=1000000000000.00
I0316 08:06:03.361439    2136 segment_rpc.go:140] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Downloaded segment dur=30.8224ms
I0316 08:06:03.361439    2136 census.go:1198] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentDownloaded... dur=30.8224ms
I0316 08:06:03.361439    2136 orchestrator.go:475] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Starting to transcode segment
I0316 08:06:03.361439    2136 orchestrator.go:463] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Creating new segment chan
I0316 08:06:03.361439    2136 orchestrator.go:608] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Starting transcode segment loop for manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf sessionID=3c256dbc
I0316 08:06:03.361439    2136 orchestrator.go:484] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Submitted segment to transcode loop
I0316 08:06:03.371522    2136 lb.go:85] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Creating transcode session for job=3c256dbc
I0316 08:06:03.371522    2136 lb.go:130] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Created transcode session for key=3c256dbc_0
I0316 08:06:03.371522    2136 lb.go:209] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode submitted for key=3c256dbc_0
I0316 08:06:03.558748    2136 orchestrator.go:567] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoding of segment took=187.2264ms
I0316 08:06:03.558748    2136 census.go:1303] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentTranscode nonce=0 seqNo=1 dur=187.2264ms trusted=true verified=true
E0316 08:06:03.558748    2136 orchestrator.go:578] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Cannot find transcoded segment for bytes=0
E0316 08:06:03.568864    2136 segment_rpc.go:234] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Could not transcode err="ZeroSegments"
I0316 08:07:03.562412    2136 lb.go:181] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=84.17.50.98 LB: Transcode loop timed out for key=3c256dbc_0
I0316 08:07:03.572414    2136 orchestrator.go:643] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Segment loop timed out; closing
I0316 08:07:03.607111    2136 lb.go:122] manifestID=b145058d-e8bb-4288-b4c3-2f612b5cd4cf seqNo=1 orchSessionID=3c256dbc clientIP=84.17.50.98 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Deleted transcode session for key=3c256dbc_0
```

Guidelines from @yondonfu for investigation:
> A few things that can/should be investigated:
> 
> - How B handles the ZeroSegment error i.e. does it treat it as a non-retryable error in which case it acknowledges that the problem is likely with the segment itself
> - See if its possible to access segment data on hosted Bs to do additional forensics on why those segments cause ZeroSegment errors

**Comment by papabear99** [comment]
These tend to come in batches.  The error above was 1 of 7 segments that I received in a row on that node.  I've seen much longer batches and across multiple regions that all happen during the same time period, sometimes lasting hours.

**Comment by Franck-UltimaRatio** [comment]
We got tons of this error this last days, specially on london and frunkfurt node fyi.

**Comment by papabear99** [comment]
These errors also appear to kill active sessions
```
I0316 11:53:40.773821    6280 segment_rpc.go:100] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Received segment dur=2s
I0316 11:53:40.775228    6280 census.go:1136] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentEmerged... duration=2
I0316 11:53:40.789280    6280 segment_rpc.go:140] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Downloaded segment dur=11.7433ms
I0316 11:53:40.790079    6280 census.go:1198] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentDownloaded... dur=11.7433ms
I0316 11:53:40.792025    6280 orchestrator.go:475] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Starting to transcode segment
I0316 11:53:40.792579    6280 orchestrator.go:484] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Submitted segment to transcode loop
I0316 11:53:40.793602    6280 lb.go:60] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Using existing transcode session for key=bd3ff7e7_0
I0316 11:53:40.794145    6280 lb.go:209] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode submitted for key=bd3ff7e7_0
I0316 11:53:40.879458    6280 orchestrator.go:567] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoding of segment took=85.8561ms
I0316 11:53:40.879968    6280 census.go:1303] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentTranscode nonce=0 seqNo=95 dur=85.8561ms trusted=true verified=true
I0316 11:53:40.880475    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoded segment profile=720p bytes=515684
I0316 11:53:40.882535    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoded segment profile=480p bytes=262636
I0316 11:53:40.883553    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=95 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoded segment profile=360p bytes=142692
I0316 11:53:40.885072    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/360p/95.ts
I0316 11:53:40.885072    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/720p/95.ts
I0316 11:53:40.895005    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/480p/95.ts
I0316 11:53:42.773857    6280 segment_rpc.go:100] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Received segment dur=2s
I0316 11:53:42.775227    6280 census.go:1136] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentEmerged... duration=2
I0316 11:53:42.778427    6280 orchestrator.go:170] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Receiving ticket sessionID=bd3ff7e7 faceValue=0.0556371564 ETH winProb=0.0000179736 ev=1000000000000.00
I0316 11:53:42.789558    6280 segment_rpc.go:140] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Downloaded segment dur=8.8966ms
I0316 11:53:42.790807    6280 census.go:1198] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentDownloaded... dur=8.8966ms
I0316 11:53:42.795571    6280 orchestrator.go:475] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Starting to transcode segment
I0316 11:53:42.796085    6280 orchestrator.go:484] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Submitted segment to transcode loop
I0316 11:53:42.796596    6280 lb.go:60] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Using existing transcode session for key=bd3ff7e7_0
I0316 11:53:42.797626    6280 lb.go:209] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode submitted for key=bd3ff7e7_0
I0316 11:53:42.886563    6280 orchestrator.go:567] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoding of segment took=89.9668ms
I0316 11:53:42.887102    6280 census.go:1303] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentTranscode nonce=0 seqNo=96 dur=89.9668ms trusted=true verified=true
I0316 11:53:42.888193    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Transcoded segment profile=720p bytes=518504
I0316 11:53:42.890398    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoded segment profile=480p bytes=269216
I0316 11:53:42.891463    6280 orchestrator.go:586] manifestID=861b842ad1apn4k3 seqNo=96 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoded segment profile=360p bytes=143444
I0316 11:53:42.893071    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/360p/96.ts
I0316 11:53:42.897050    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/720p/96.ts
I0316 11:53:42.902458    6280 player.go:105] LPMS got HTTP request @ /stream/bd3ff7e7/480p/96.ts
I0316 11:53:53.508083    6280 segment_rpc.go:100] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Received segment dur=3.829s
I0316 11:53:53.508641    6280 census.go:1136] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentEmerged... duration=3.8289999999999997
I0316 11:53:53.510220    6280 orchestrator.go:170] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Receiving ticket sessionID=9b90c27c faceValue=0.0556371564 ETH winProb=0.0000179736 ev=1000000000000.00
I0316 11:53:53.535803    6280 segment_rpc.go:140] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Downloaded segment dur=25.0541ms
I0316 11:53:53.536311    6280 census.go:1198] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Logging SegmentDownloaded... dur=25.0541ms
I0316 11:53:53.540576    6280 orchestrator.go:475] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Starting to transcode segment
I0316 11:53:53.541205    6280 orchestrator.go:463] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Creating new segment chan
I0316 11:53:53.541255    6280 orchestrator.go:608] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Starting transcode segment loop for manifestID=daff8019-9494-438e-86f6-22501dd72e7a sessionID=9b90c27c
I0316 11:53:53.541942    6280 orchestrator.go:484] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Submitted segment to transcode loop
I0316 11:53:53.543010    6280 lb.go:85] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Creating transcode session for job=9b90c27c
I0316 11:53:53.543010    6280 lb.go:130] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Created transcode session for key=9b90c27c_0
I0316 11:53:53.543525    6280 lb.go:209] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode submitted for key=9b90c27c_0
I0316 11:53:53.717174    6280 orchestrator.go:567] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Transcoding of segment took=174.1747ms
I0316 11:53:53.727216    6280 census.go:1303] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Logging SegmentTranscode nonce=0 seqNo=1 dur=174.1747ms trusted=true verified=true
E0316 11:53:53.727216    6280 orchestrator.go:578] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E clientIP=195.181.174.39 Cannot find transcoded segment for bytes=0
E0316 11:53:53.727216    6280 segment_rpc.go:234] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Could not transcode err="ZeroSegments"
I0316 11:54:42.894645    6280 orchestrator.go:643] manifestID=861b842ad1apn4k3 seqNo=28 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Segment loop timed out; closing
I0316 11:54:42.894645    6280 lb.go:181] manifestID=861b842ad1apn4k3 seqNo=28 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode loop timed out for key=bd3ff7e7_0
I0316 11:54:42.969518    6280 lb.go:122] manifestID=861b842ad1apn4k3 seqNo=28 orchSessionID=bd3ff7e7 clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Deleted transcode session for key=bd3ff7e7_0
I0316 11:54:53.728724    6280 lb.go:181] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Transcode loop timed out for key=9b90c27c_0
I0316 11:54:53.728724    6280 orchestrator.go:643] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E Segment loop timed out; closing
I0316 11:54:53.793771    6280 lb.go:122] manifestID=daff8019-9494-438e-86f6-22501dd72e7a seqNo=1 orchSessionID=9b90c27c clientIP=195.181.174.39 sender=0xc3c7c4C8f7061B7d6A72766Eee5359fE4F36e61E LB: Deleted transcode session for key=9b90c27c_0
```

**Comment by leszko** [comment]
Some observations:
- The segment is retried in B, sent multiple times with the same outcome `ZeroSegments`
- We experience exactly the same in the Livepeer Inc Orchestrators, so there is something fishy with the segment itself
- The transcoded segment has size `0`

It'd would be great to grab the source segment and investigate it locally. Working on it...

**Comment by AlexKordic** [comment]
Single IDR frame segment as input causes `ZeroSegments` error. The proper output would be single frame segment output.

**Comment by papabear99** [comment]
Is it possible to reject these segments and throw the error at the B instead of sending them down network to be transcoded?  Seems like it would benefit Broadcasters to know why their segments can't be transcoded.  My node in the Netherlands has received nothing but these `ZeroSegment` sessions for the last hour+

**Comment by AlexKordic** [comment]
> Is it possible to reject these segments and throw the error at the B instead of sending them down network to be transcoded? Seems like it would benefit Broadcasters to know why their segments can't be transcoded. My node in the Netherlands has received nothing but these `ZeroSegment` sessions for the last hour+

Yes. My understanding is: this is still valid transcoding input. Is that correct @leszko ?

**Comment by yondonfu** [comment]
@AlexKordic 

> Single IDR frame segment as input causes ZeroSegments error. The proper output would be single frame segment output.

Interesting! Sounds like the current output for a single IDR frame source segment triggers the `ZeroSegments` error either because [the output is nil or because the output has less than 25 bytes](https://github.com/livepeer/go-livepeer/blob/6872dbab566ec3a0b17f1a1b573ddf9401a5a574/core/orchestrator.go#L577). Awhile back LPMS couldn't handle segments with very few frames (i.e. 1 frame) as documented in https://github.com/livepeer/lpms/issues/168 but this was supposed to be fixed in https://github.com/livepeer/lpms/pull/189. So, something about this type of segment is likely not being handled well by that fix.

A single frame output for a single frame input sounds like the right behavior so if the transcoding pipeline can be tweaked to support that behavior for this type of segment that seems ideal. If that is not possible, then the next best thing seems to be to detect this type of segment to reject it earlier if possible since no valid output would be generated. We can likely start by just determining why this type of segment is causing problems in the first place.

**Comment by leszko** [comment]
I think the one-frame segment sounds like a correct input, so we should handle it correctly in LPMS. @AlexKordic created a GH Issue for it: https://github.com/livepeer/lpms/issues/311

The other problem is that we need to think how to approach the problem if people start sending a lot of one-frame streams. The cost of processing such streams is very high, but you get very little payment for it. I created a separate GH Issue to address it and we should take it into our roadmap at some point. https://github.com/livepeer/go-livepeer/issues/2335

Closing this GH Issue since solving the issue here is actually related to LPMS.
